### PR TITLE
Fixes bug where slashes are removed from the vault paths

### PR DESCRIPTION
**Pull request by allez-allez-allez**
This patch fixes the bug #10  where slashes in the path are being removed. Also, did some cleaning up to the changes result of the salt `vault.synced` states:

An example of the new change results:
```yaml
 Function: vault.secret_engines_synced
      Result: True
     Comment: 
     Started: 21:46:55.872968
    Duration: 30.755 ms
     Changes:   
              ----------
              new:
                  ----------
                  cubbyhole/:
                      ----------
                      accessor:
                          cubbyhole_628b625a
                      config:
                          ----------
                          default_lease_ttl:
                              0
                          force_no_cache:
                              False
                          max_lease_ttl:
                              0
                      description:
                          per-token private secret storage
                      local:
                          True
                      options:
                          None
                      seal_wrap:
                          False
                      type:
                          cubbyhole
                      uuid:
                          8460db26-1c1e-74b4-ff72-e509c13e9c9d
```

### CODE REVIEW

File: salt/_modules/vault.py [hunk 1]
```diff
@@ -189,20 +195,22 @@ def push_policies(self, client, remote_policies, local_policies, ret):
...
                 log.debug('Policy "%s" has been updated.', policy["name"])
+                existing_policies.append(policy["name"])
             else:
                 new_policies.append(policy["name"])
                 log.debug('Policy "%s" has been created.', policy["name"])
 
         # Build return object
-        ret['changes']['old'] = remote_policies
-        ret['changes']['new'] = new_policies or "No changes"
+        ret['changes']['updated'] = existing_policies or "None"
```
[the remark this answers is not in the document]
Comment (allez-allez-allez) [reply salt/_modules/vault.py]: makes sense. I update the `changes` output to displaying the new changes. Simply calling the vault APIs to return the latest information from Vault for secrets engines, auth methods, policies, and audit devices...

**Comment by lorengordon** [comment]
Ok, so when the changes are a dictionary, salt has a deep diff utility you can use to display what actually changed rather than always outputting the entire dictionary... Here's an example...

* https://github.com/saltstack/salt/pull/40235/files

**Comment by allez-allez-allez** [comment]
alright, incorporate `salt.utils.dictdiffer.deep_diff` to the project. `changes` results look much cleaner than before. Ready for another look. Thanks!

**Comment by lorengordon** [comment]
Let's do it

**Pull request merged**